src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case ('addProperties'):
+        Object.assign(state, action.extraData);
+        break;
+
+      case ('clear'):
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      case ('removeProperties'):
+        for (const keyToRemove of action.keysToRemove) {
+          delete state[keyToRemove];
+        }
+        break;
+
+      default:
+        return state;
```

Comment:
perhaps, it'd be better to throw an error
however, we've not been told what to do, so OK 🙂